### PR TITLE
Geb HVAC - Single Speed and Two Speed Systems

### PR DESCRIPTION
**Pull request by yzhou601**
## Pull Request Description
Single Speed features:

1.  On-off Deadband thermostat
2. Start-up Loss

Two Speed features:

1. On-off Deadband thermostat
2. Realistic Staging

Alternative of #458 

## Checklist

Not all may apply:

- [x] EPvalidator.xml has been updated
- [x] Tests (and test files) have been updated
- [x] Documentation has been updated
- [x] Changelog has been updated
- [x] `openstudio tasks.rb update_measures` has been run
- [x] No unexpected regression test changes on CI (checked comparison artifacts)

### CODE REVIEW

**Comment by yzhou601** [comment]
There's something weird, CI complains an extra warning:

```
   ** Warning ** The following Report Variables were requested but not generated -- check.rdd file
   **   ~~~   ** Either the IDF did not contain these elements, the variable name is misspelled,
   **   ~~~   ** or the requested variable is an advanced output which requires Output : Diagnostics, DisplayAdvancedReportVariables;
   ************* Key=CENTRAL AC CLG COIL, VarName=COOLING COIL CRANKCASE HEATER ELECTRICITY ENERGY, Frequency=
```

I looked at the base-hvac-central-ac-only-2-speed.xml test file where the realistic staging test file is based from, and there's no warning thrown. And I compared these two idf files, all the changes seems unrelated to the crankcase heater. But IDD file for the realistic staging model no longer lists the crankcase energy variable, while that for the base central ac model does. I am not sure if it is caused by the speed override actuator or something else.
[in-twospeedac-realistic.txt](https://github.com/NREL/OpenStudio-HPXML/files/9976549/in-twospeedac-realistic.txt)
[in-twospeedac.txt](https://github.com/NREL/OpenStudio-HPXML/files/9976552/in-twospeedac.txt)

**Comment by yzhou601** [comment]
I added a dumb EMS program using the following actuator, where I simply set the speed level at 1, the COOLING COIL CRANKCASE HEATER ELECTRICITY ENERGY variable is also removed from idd file:

```
EnergyManagementSystem:Actuator,
  central_ac_unitary_system_speed_override, !- Name
  central ac unitary system,              !- Actuated Component Unique Name
  Coil Speed Control,                     !- Actuated Component Type
  Unitary System DX Coil Speed Value;     !- Actuated Component Control Type
```

```
EnergyManagementSystem:Program,
  central_ac_unitary_system_realistic_cycling, !- Name
  Set central_ac_unitary_system_speed_override = 1; !- Program Line 15
```

So it should be a bug on E+ side for this speed level actuator. One path forward is to make this warning an exclusion for the realistic staging test file and document it as a known issue. Then we need to fix E+ to report crankcase heater energy.

**Comment by shorowit** [comment]
> I added a dumb EMS program using the following actuator, where I simply set the speed level at 1, the COOLING COIL CRANKCASE HEATER ELECTRICITY ENERGY variable is also removed from idd file:
> 
> ```
> EnergyManagementSystem:Actuator,
>   central_ac_unitary_system_speed_override, !- Name
>   central ac unitary system,              !- Actuated Component Unique Name
>   Coil Speed Control,                     !- Actuated Component Type
>   Unitary System DX Coil Speed Value;     !- Actuated Component Control Type
> ```
> 
> ```
> EnergyManagementSystem:Program,
>   central_ac_unitary_system_realistic_cycling, !- Name
>   Set central_ac_unitary_system_speed_override = 1; !- Program Line 15
> ```
> 
> So it should be a bug on E+ side for this speed level actuator. I couldn't think of any reason why it is no longer reporting any crankcase energy. One path forward is to make this warning an exclusion for the realistic staging test file and document it as a known issue. Then we need to fix E+ to report crankcase heater energy.

That seems reasonable to me. Please file an E+ GitHub bug report on this too.

**Comment by yzhou601** [comment]
We might not include this PR in the 1.5.0 release due to the [EnergyPlus issue](https://github.com/NREL/EnergyPlus/issues/9732).

**Comment by yzhou601** [comment]
@jmaguire1 Energyplus bugfix tested for previous test files generated from this PR. Previous issues resolved in that branch. A few action items:
 - [ ] We implemented 4 mins instead of 5 mins for the time based staging logic, should change to 5 mins
 - [ ] Unitary system sensible cooling energy is much lower than coil sensible cooling energy, even with after subtracting fan heat, need to run one without EMS program to isolate the cause of the issue.
 - [ ] Give updates of the test runs to E+ team, with a summary comment on PR.

**Comment by yzhou601** [comment]
Based on the latest discussions with @jmaguire1 
Next steps for two speed systems:
  - [ ] Check if we need better start-up time assumption for two speed systems. @jmaguire1 
  - [ ] Fix the current ci error caused by merging @yzhou601

File: workflow/sample_files/base-hvac-realistic-control-2-speed-central-ac.xml [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+    <XMLType>HPXML</XMLType>
+    <XMLGeneratedBy>tasks.rb</XMLGeneratedBy>
+    <CreatedDateAndTime>2000-01-01T00:00:00-07:00</CreatedDateAndTime>
+    <Transaction>create</Transaction>
+  </XMLTransactionHeaderInformation>
+  <SoftwareInfo>
+    <extension>
+      <SimulationControl>
+        <Timestep>1</Timestep>
+        <TemperatureCapacitanceMultiplier>4.0</TemperatureCapacitanceMultiplier>
```
Comment (yzhou601) [inline workflow/sample_files/base-hvac-realistic-control-2-speed-central-ac.xml:13]: update this to 7

File: workflow/tests/base_results/results_bills.csv [hunk 2]
```diff
@@ -1,437 +1,425 @@
...
-base-dhw-indirect-detailed-setpoints.xml,1425.13,144.0,786.31,0.0,930.31,144.0,350.82,494.82,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
+base-appliances-coal.xml,1625.92,144.0,1105.99,0.0,1249.99,144.0,158.94,302.94,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,72.99,72.99,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
```
Comment (yzhou601) [inline workflow/tests/base_results/results_bills.csv:2]: Why all the rows in results_bills.csv are changed , while results_workflow_simulation1.csv and results_workflow_simulation1_bills.csv are not changed except new rows are added.

Comment (shorowit) [reply workflow/tests/base_results/results_bills.csv:2]: Some of these test artifacts were renamed/deleted, but someone keeps resolving conflicts wrong and bringing these files back. For example, I see that this branch is reintroducing the `results.csv` and `results_hvac_sizing.rb` files, which don't exist on master.

`results_bill.csv` also shouldn't exist on mster, I will delete it (again).

Comment (shorowit) [reply workflow/tests/base_results/results_bills.csv:2]: [Deleted results_bills.csv on master](https://github.com/NREL/OpenStudio-HPXML/commit/80ae0a54a991d3bd5296381232b7ecefb834aa9b)

File: Changelog.md [hunk 3]
```diff
@@ -179,6 +179,8 @@ __New Features__
 - Stochastic schedules for garage lighting and TV plug loads now use interior lighting and miscellaneous plug load schedules, respectively.
 - Performance improvement for HPXML files w/ large numbers of `Building` elements.
 - Weather cache files (\*foo-cache.csv) are no longer used/needed.
+- **Research Feature**: Allows modeling on/off thermostat deadband with start-up degradation for single speed central ac and single speed ashp systems. Currently only supports homes with at most one cooling system serving 100% cooling loads, and one heating system serving 100% heating loads (including heat pumps). An on/off thermostat deadband temperature is a temperature difference between cut-in and cut-out temperature for HVAC operations, applied to both heating and cooling.
+- **Research Feature**: Allows modeling time-based realistic staging for two speed DX systems (central ac and ASHP). Assumes 5 minutes before transitioning to a higher speed.
```
Comment (shorowit) [inline Changelog.md:183]: Needs to be moved up to OS-HPXML v1.8. Also, we should make these notes consistent with the variable-speed GEB item. And I would shorten the descriptions -- if any of the detail is important, it should be in the documentation, not the changelog.

File: workflow/tests/base_results/results_hvac_sizing.csv
Comment (shorowit) [inline workflow/tests/base_results/results_hvac_sizing.csv:1]: This file should be deleted.

**Comment by shorowit** [comment]
Discussion notes:
1. Create new `/HPXML/Building/BuildingDetails/BuildingSummary/extension/HVACGEBControl` element with `UseRealisticCyclingModel` and a HP backup capacity increment input.
2. Combine this PR with the HP backup staging PR.
3. Add **Research Feature** to these features and existing ones (variable-speed capacity ratios, HPWH operating mode schedule, conventional storage WH tank type, temperature capacitance multiplier (TCM), etc.) in the documentation.
4. Make sure we give a warning if TCM is 1 when used with some of these features.

**Comment by yzhou601** [comment]
This is combined into #1639

**Pull request closed without merging**